lib/sparkql/function_resolver.rb

```diff
@@ -118,6 +122,13 @@ def call()
   protected 
   
   # Supported function calls
+
+  def regex(regular_expression)
+    {
+      :type => :character,
+      :value => regular_expression
```

Comment:
I think we want to validate here that the parameter is, in fact, a correct regular expression.  For example:

```
irb(main):005:0> Regexp.new("[1234")
RegexpError: premature end of char-class: /[1234/
        from (irb):5:in `initialize'
        from (irb):5:in `new'
        from (irb):5
        from /opt/ruby/bin/irb:12:in `<main>'
```

Also, just floating this out there, what would you guys think about a literally returning `Regex.new`?  I'm not sure I'm sold on that, since we don't have a native sparkql regex type.